api/v1alpha1/osbuild_types.go

```diff
@@ -25,19 +25,45 @@ import (
 
 // OSBuildSpec defines the desired state of OSBuild
 type OSBuildSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Details defines what to build
+	Details BuildDetails `json:"details"`
 
-	// Foo is an example field of OSBuild. Edit osbuild_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// TriggeredBy explains what triggered the build out
+	TriggeredBy TriggeredBy `json:"config"`
 }
 
+// +kubebuilder:validation:Enum=UpdateCR;Webhook
+type TriggeredBy string
+
```

Comment:
Right now we are supposed to support only 2 types - rest API(webhook) and update the CR and trigger the reconcile 
If you think about more options please share and I will add it to the enum

Reply:
None at the moment. Just wondering if you were going to create the instances later on or still considering which ones. Here it would be a good opportunity to make use of the types rather than a hardcoded string, what do you think?
https://github.com/project-flotta/osbuild-operator/blob/1baa26ea81d7db314457645dd86cdb74b112e4d0/controllers/osbuildconfig_controller.go#L88-L91